services/frontend/src/scenes/Profile/connect.js

```diff
@@ -1,11 +1,35 @@
 import {connect} from "react-redux";
+import {matchPath} from "react-router";
 import {crossSliceSelectors} from "store/";
-import {profilesRequestsSlice} from "store/slices";
+import {reduceLoadingStates} from "utils/helperFunctions";
+import {profilesRequestsSlice, projectsRequestsSlice, skillsRequestsSlice} from "store/slices";
+import ScreenUrls from "utils/screenUrls";
 
-const mapStateToProps = (state) => ({
-    isLoading: profilesRequestsSlice.fetchAll.selectors.getLoading(state),
-    projects: crossSliceSelectors.getProjectsForUser(state),
-    profile: crossSliceSelectors.getUserProfile(state)
-});
+const mapStateToProps = (state, props) => {
+    const mappedState = {};
+    mappedState.isLoading = reduceLoadingStates([
+        profilesRequestsSlice,
+        projectsRequestsSlice,
+        skillsRequestsSlice
+    ], state);
+
+    if (matchPath(props.match.path, ScreenUrls.PROFILE)) {
```

Comment:
I went with creating the selector factory since I haven't done that yet,
```js
const isMatchingRoute = (route) => createSelector(
    [createMatchSelector(route)],
    (match) => match
);
```
and then called it as:
```js
if (crossSliceSelectors.isMatchingRoute(ScreenUrls.PROFILE)(state)) {
```

Reply:
Isn't that selector redundant? Wouldn't 

```js
const isMatchingRoute = (route) => createMatchSelector(route);
```

be the same thing, with one less layer?

Reply:
In fact, you could literally just use

```js
const isMatchingRoute = createMatchSelector
```

i.e. you don't even need a new selector, you can just use `createMatchSelector` directly...

:man_facepalming: on my part for not realizing that.

Reply:
Yep, that simplifies it significantly.

Swapped it over to your second comment.